conans/test/functional/devlayout/devlayout_load_test.py

```diff
@@ -0,0 +1,238 @@
+
+        client.run("create . lib/1.0@")
+        self.assertIn("Created package revision", client.out)
+
+        # Virtual load
+        client.run("install lib/1.0@")
+
+    def test_text_layout_load(self):
+        """Declare the layout using the layout text attribute"""
+        client = TestClient()
+        conanfile = textwrap.dedent("""
+           from conans import ConanFile, CMake
+    
+           class Pkg(ConanFile):
+               settings = "os", "compiler", "arch", "build_type"
+               layout = "cmake"
+               
+               def assert_layout(self):
+                   assert self.layout == "cmake"
+                   assert self.lyt.build == "build"
+                   assert self.lyt.src == ""
+    
+               def build(self):
+                   self.assert_layout()
+    
+               def package(self):
+                   self.assert_layout()
+    
+               def package_info(self):
+                   self.assert_layout()
+               """)
+        client.save({"conanfile.py": conanfile})
+
+        client.run("create . lib/1.0@")
+        self.assertIn("Created package revision", client.out)
+
+        # Virtual load
+        client.run("install lib/1.0@")
+
+    def load_layout_py_local_methods_test(self):
+        client = TestClient()
+        conanfile = textwrap.dedent("""
+           from conans import ConanFile, CMake
+
+           class Pkg(ConanFile):
+               settings = "os", "compiler", "arch", "build_type"
+               layout = "cmake"
+
+               def build(self):
+                   assert self.lyt.build == "overwritten_build"
+                   assert self.lyt.src == "overwritten_src"
+                   self.output.warn("Here, building")
+               """)
+        override_layout = textwrap.dedent("""
+        from conans import Layout
+        
+        def layout(self):
+            ly = Layout(self)
+            ly.build = "overwritten_build"
+            ly.src = "overwritten_src"
+            return ly
+        """)
+        client.save({"conanfile.py": conanfile, LAYOUT_PY: override_layout})
+        client.run("install .")
+        # If not specified in the layout, layout.build_installdir is defauted to layout.build
+        client.run("build . -if=overwritten_build")
+        self.assertIn("Here, building", client.out)
+
+    def not_load_layout_create_test(self):
+        """The LAYOUT_PY file is not used with a Conan create because the package is in the cache
+           and this is only intended to work only for packages being edited (local methods +
+           editables) Otherwise a package in the cache would have been generated with a "patchy"
+           conanfile.
+           """
+        client = TestClient()
+        conanfile = textwrap.dedent("""
+           from conans import ConanFile, CMake
+
+           class Pkg(ConanFile):
+               settings = "os", "compiler", "arch", "build_type"
+               layout = "cmake"
```

Comment:
Make sure to ``exports`` the layout file, so it is also in the cache, will be loaded then? Can be exported by accident?

Reply:
we should block exporting it. missing test.